src/Donations/Micros/OnBehalf.tsx

```diff
@@ -0,0 +1,141 @@
+import { Grid } from "@material-ui/core";
+import { useTranslation } from "next-i18next";
+import { FC, useContext, useState } from "react";
+import { useForm } from "react-hook-form";
+import MaterialTextField from "src/Common/InputTypes/MaterialTextField";
+import ToggleSwitch from "src/Common/InputTypes/ToggleSwitch";
+import { QueryParamContext } from "src/Layout/QueryParamContext";
+
+const OnBehalf: FC = () => {
+  const { t } = useTranslation("common");
+  const { onBehalf, setOnBehalf, onBehalfDonor, setOnBehalfDonor } =
+    useContext(QueryParamContext);
+
+  const [shouldNotify, setShouldNotify] = useState(false);
+
+  const defaultValues = {
+    firstName: onBehalfDonor.firstName,
+    lastName: onBehalfDonor.lastName,
+    email: onBehalfDonor.email,
+  };
+
+  const { register, errors, handleSubmit, reset } = useForm({
+    mode: "all",
+    defaultValues,
+  });
+
+  const onSubmit = (data) => {
+    setOnBehalfDonor(data);
+  };
+
+  const resetOnBehalfForm = () => {
+    const defaultValues = {
+      firstName: "",
+      lastName: "",
+      email: "",
+    };
+    setOnBehalfDonor(defaultValues);
+    setOnBehalf(false);
+    reset(defaultValues);
+  };
+
+  return onBehalfDonor.firstName === "" ? (
+    <div className="on-behalf-container mt-10">
+      <div className="on-behalf-container-toggle">
+        <p>{t("donationOnBehalfText")}</p>
+        <ToggleSwitch
+          checked={onBehalf}
+          onChange={() => setOnBehalf((onBehalf) => !onBehalf)}
+        />
+      </div>
+      {onBehalf && onBehalfDonor.firstName === "" ? (
```

Comment:
- [ ] There is no way to cancel the input of doing the donation on behalf of someone (same as for dedicate the donation to someone). This is a UI issue. A simple back button or close icon would be nice to have?

<img width="831" alt="Bildschirmfoto 2022-03-13 um 11 22 42" src="https://user-images.githubusercontent.com/1532418/158055157-d45bcca4-30b7-4740-b43c-87424d62a2b9.png">

Reply:
At least after entering some random name, you can delete it again.